cli/main.rs

```diff
@@ -0,0 +1,39 @@
+use std::env;
+
+use clap::Parser;
+use sp_runtime::{generic, traits::BlakeTwo256};
+use subxt::PolkadotConfig;
+use try_runtime_core::commands::TryRuntime;
+
+#[subxt::subxt(
```

Comment:
any thoughts about how this would work against arbitrary substrate-based chains? I am sure I was told subxt supports this, but don't have it off the top of my head.

Reply:
This macro generates all of the static interface to talk to the node that the metadata comes from. Currently, as long as other nodes support the exact same interface that the metadata was generated against, it'll mostly Just Work. Subxt will take care of things like call indexes and pallet indexes being different across nodes.

subxt has validation logic which will do a runtime check when you use this interface to verify that the calls etc you use line up with the node you're trying to talk to (this validation can be switched off if you want to yolo it)

An in flight PR moves the static interface to use new encode and decode machinery (see scale-encode and scale-decode crates) which loosens the requirements on the static code the macro generates being exactly identical to the metadata, basically giving more leeway for things to keep working if the interface does change a little in certain ways (though the validation logic still will warn if things change and needs opting out of if interfaces do change in a breaking way)

So yup, it'll work cross chain as long as the pallet interfaces you use line up.